app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,63 @@
+class CategoriesController < ApplicationController
+  before_action :set_category, only: %i[show edit update destroy]
+
+  def index
+    @categories = Category.includes(:buys).all
+    @categories.each do |category|
+      total = 0
+      category.buys.each do |buy|
+        total += buy.amount
+      end
+      category.total = total
+    end
+  end
```

Comment:
**OPTIONAL** -You did well implementing this, and I know it is one you are proud of 👏 but this is not considered a good practice as you are adding a new attribute to your category object. Since you are looking to get the total for all transactions belonging to a category, in your index file for your category view page, when looping through all categories, you can get the sum of transactions for each category by doing this:

```ruby 
      category.buys.sum(:amount)
```
Doing this would help make your code faster because you do not have to implement this loop in your controller. Please fix this 👍 👍 . It also helps makes your controller look simple, and makes your code cleaner

Reply:
done